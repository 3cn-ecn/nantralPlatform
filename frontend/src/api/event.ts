import axios from 'axios';
import {
  EventProps,
  eventToCamelCase,
  eventsToCamelCase,
} from '../Props/Event';

export async function getEvents(options?: {
  orderBy?: string[] | string;
  fromDate?: Date;
  toDate?: Date;
  limit?: number;
}): Promise<EventProps[]> {
  const { data } = await axios.get('/api/event/', {
    params: {
      from_date: options?.fromDate,
      to_date: options?.toDate,
      order_by: Array.isArray(options?.orderBy)
        ? options?.orderBy?.join(',')
        : options?.orderBy,
      limit: options?.limit,
    },
  });
  eventsToCamelCase(data.results);
  return data.results;
}

export async function getEvent(id: number): Promise<EventProps> {
  const { data } = await axios.get(`/api/event/${id}/`);
  eventToCamelCase(data);
  return data;
}
