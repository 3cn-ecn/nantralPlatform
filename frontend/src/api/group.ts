import axios from 'axios';
import { GroupProps, SimpleGroupProps } from '../Props/Group';
import { Page } from '../Props/pagination';

export async function getMyGroups() {
  return getGroups({ isMember: true });
}

export async function getGroups(
  options: {
    isMember?: boolean;
    simple?: boolean;
    limit?: number;
    order?: string;
    admin?: boolean;
  } = {}
): Promise<Array<SimpleGroupProps | GroupProps>> {
  return axios
    .get<Page<SimpleGroupProps | GroupProps>>('/api/group/group/', {
      params: {
        is_member: options.isMember,
        simple: options.simple,
        order: options.order,
        is_admin: options.admin,
      },
    })
    .then((res) => res.data.results);
}

export async function getGroup(
  slug: string,
  options?: { simple?: boolean }
): Promise<GroupProps> {
  return axios
    .get<GroupProps>(`/api/group/group/${slug}`, {
      params: { simple: options.simple },
    })
    .then((res) => res.data);
}
