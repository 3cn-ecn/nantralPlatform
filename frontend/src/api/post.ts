import axios from 'axios';
import { PostProps, convertPostFromPythonData } from '../Props/Post';

export async function getPost(id: number) {
  return axios
    .get<PostProps>(`/api/post/${id}/`)
    .then((res) => res.data)
    .then((data) => convertPostFromPythonData(data));
}

export async function getPosts(
  options: {
    pinned?: boolean;
    fromDate?: Date;
    toDate?: Date;
    limit?: number;
  } = {}
) {
  return axios
    .get<PostProps[]>(`/api/post/`, {
      params: {
        pinned: options.pinned || undefined,
        from_date: options.fromDate,
        to_date: options.toDate,
        limit: options.limit,
      },
    })
    .then((res) => convertPostFromPythonData(res.data));
}
