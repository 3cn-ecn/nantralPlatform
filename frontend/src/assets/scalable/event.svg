<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<!-- Generator: Adobe Illustrator 25.0.0, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->

<svg
   version="1.1"
   id="Calque_1"
   x="0px"
   y="0px"
   viewBox="0 0 24 24"
   style="enable-background:new 0 0 24 24;"
   xml:space="preserve"
   sodipodi:docname="event.svg"
   inkscape:version="1.2.2 (b0a84865, 2022-12-01)"
   xmlns:inkscape="http://www.inkscape.org/namespaces/inkscape"
   xmlns:sodipodi="http://sodipodi.sourceforge.net/DTD/sodipodi-0.dtd"
   xmlns="http://www.w3.org/2000/svg"
   xmlns:svg="http://www.w3.org/2000/svg"><defs
   id="defs17314" /><sodipodi:namedview
   id="namedview17312"
   pagecolor="#505050"
   bordercolor="#eeeeee"
   borderopacity="1"
   inkscape:showpageshadow="0"
   inkscape:pageopacity="0"
   inkscape:pagecheckerboard="0"
   inkscape:deskcolor="#505050"
   showgrid="false"
   inkscape:zoom="9.8333333"
   inkscape:cx="11.949153"
   inkscape:cy="12"
   inkscape:window-width="1440"
   inkscape:window-height="872"
   inkscape:window-x="0"
   inkscape:window-y="28"
   inkscape:window-maximized="0"
   inkscape:current-layer="Calque_1" />
<style
   type="text/css"
   id="style17296">
	.eventst0{fill:#FFFFFF;stroke:#B6B7B7;stroke-width:0.25;stroke-miterlimit:10;}
	.eventst1{fill:url(#SVGID_1_);stroke:#B6B7B7;stroke-width:0.25;stroke-miterlimit:10;}
	.eventst2{fill:#DC3545;}
</style>

<linearGradient
   id="SVGID_1_"
   gradientUnits="userSpaceOnUse"
   x1="4"
   y1="19.78"
   x2="20"
   y2="19.78"
   gradientTransform="matrix(1,0,0,-1,0,26)">
	<stop
   offset="0"
   style="stop-color:#B6B7B7"
   id="stop17300" />
	<stop
   offset="1"
   style="stop-color:#E0E0DF"
   id="stop17302" />
</linearGradient>


<g
   id="g17800"
   transform="matrix(1.476923,0,0,1.476923,-5.7230767,-5.7230767)"><path
     class="eventst0"
     d="m 20,8.4 v 9.8 c 0,1 -0.8,1.8 -1.8,1.8 v 0 H 5.8 C 4.8,20 4,19.2 4,18.2 v 0 -9.8"
     id="path17298" /><path
     class="eventst1"
     d="M 4,8.4 V 5.8 C 4,4.8 4.8,4 5.8,4 h 12.4 c 1,0 1.8,0.8 1.8,1.8 0,0 0,0 0,0 v 2.6"
     id="path17305"
     style="fill:url(#SVGID_1_)" /><circle
     class="eventst2"
     cx="12"
     cy="17"
     r="1"
     id="circle17307" /><path
     class="eventst2"
     d="m 11.3,14.6 -1,-3.6 c -0.1,-0.4 0.1,-0.8 0.5,-0.9 0.1,0 0.1,0 0.2,0 h 2 c 0.4,0 0.7,0.3 0.7,0.7 0,0.1 0,0.1 0,0.2 l -1,3.6 C 12.6,15 12.2,15.2 11.8,15.1 11.6,15 11.4,14.8 11.3,14.6 Z"
     id="path17309" /></g>
</svg>
