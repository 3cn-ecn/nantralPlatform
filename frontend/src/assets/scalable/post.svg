<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<svg
   viewBox="0 0 24 24"
   version="1.1"
   id="svg1135"
   sodipodi:docname="post.svg"
   inkscape:version="1.2.2 (b0a84865, 2022-12-01)"
   xmlns:inkscape="http://www.inkscape.org/namespaces/inkscape"
   xmlns:sodipodi="http://sodipodi.sourceforge.net/DTD/sodipodi-0.dtd"
   xmlns="http://www.w3.org/2000/svg"
   xmlns:svg="http://www.w3.org/2000/svg">
  <sodipodi:namedview
     id="namedview1137"
     pagecolor="#505050"
     bordercolor="#eeeeee"
     borderopacity="1"
     inkscape:showpageshadow="0"
     inkscape:pageopacity="0"
     inkscape:pagecheckerboard="0"
     inkscape:deskcolor="#505050"
     showgrid="false"
     inkscape:zoom="9.8333333"
     inkscape:cx="9.8644068"
     inkscape:cy="12"
     inkscape:window-width="1440"
     inkscape:window-height="872"
     inkscape:window-x="0"
     inkscape:window-y="28"
     inkscape:window-maximized="0"
     inkscape:current-layer="svg1135" />
  <defs
     id="defs1127">
    <linearGradient
       id="a"
       x1="6"
       y1="17"
       x2="9"
       y2="14"
       gradientUnits="userSpaceOnUse">
      <stop
         offset="0"
         stop-color="#b6b7b7"
         id="stop1122" />
      <stop
         offset="1"
         stop-color="#e0e0df"
         id="stop1124" />
    </linearGradient>
  </defs>
  <g
     id="g1259"
     transform="matrix(1.5927288,0,0,1.5927288,-7.4121135,-6.9177852)">
    <path
       d="m 13,19 -3,-3 a 1.42,1.42 0 0 0 -2,0 v 0 l 3,3 a 1.42,1.42 0 0 0 2,0 z"
       style="fill:#b6b7b7"
       id="path1129" />
    <path
       d="M 10.69,4.62 7,12 l 4,4 7.38,-3.69 a 0.5,0.5 0 0 0 0.13,-0.8 l -7,-7 a 0.5,0.5 0 0 0 -0.82,0.11 z"
       style="fill:#dc3545"
       id="path1131" />
    <path
       d="m 7,12 c 0,0 -2,4 -1,5 1,1 5,-1 5,-1"
       style="fill:url(#a)"
       id="path1133" />
  </g>
</svg>
