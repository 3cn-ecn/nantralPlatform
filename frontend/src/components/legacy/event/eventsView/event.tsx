import * as React from 'react';
import { EventInfos } from './interfaces';
import { ParticipateButton } from './participateButton';
import { Card } from 'react-bootstrap';
import { getDate, getHour } from './utils';

export function Event(props): JSX.Element {
  const urls = props.urls;
  const eventInfos: EventInfos = props.eventInfos;

  return (
    <Card border={eventInfos.color} className="mb-2">
      <Card.Img
        variant="top"
        src={eventInfos.image}
        onClick={(e) => {
          window.location.href = eventInfos.get_absolute_url;
        }}
        style={{ cursor: 'pointer' }}
      />
      <Card.Body as="a" href={eventInfos.get_absolute_url} className="pb-0 ">
        <small className="text-uppercase card-date">
          {getDate(eventInfos.date)}
        </small>
        <Card.Title as="h4">{eventInfos.title}</Card.Title>
        <Card.Subtitle className="mb-3">
          {getHour(eventInfos.date)} • {eventInfos.location} •{' '}
          {eventInfos.get_group_name}
        </Card.Subtitle>
      </Card.Body>
      <Card.Body className="pt-0">
        <ParticipateButton
          number_of_participants={eventInfos.number_of_participants}
          urls={urls}
          eventInfos={eventInfos}
        />
      </Card.Body>
    </Card>
  );
}
