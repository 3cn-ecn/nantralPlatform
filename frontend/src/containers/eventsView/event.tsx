import * as React from "react";
import { EventInfos } from "./interfaces";
import Truncate from "react-truncate";
import { cardStyle, eventLink } from "./styles";
import { ParticipateButton } from "./participateButton";

var dayjs = require("dayjs");
var isToday = require("dayjs/plugin/isToday");
dayjs.extend(isToday);
var isTomorrow = require("dayjs/plugin/isTomorrow");
dayjs.extend(isTomorrow);
require("dayjs/locale/fr");
dayjs.locale("fr");

export function Event(props): JSX.Element {
  const urls = props.urls;
  const eventInfos: EventInfos = props.eventInfos;

  return (
    <div className={`card pt-0 bg-${eventInfos.color}`} style={cardStyle}>
      <div className="card-body">
        <a
          href={eventInfos.get_absolute_url}
          style={eventLink}
          className="mb-1"
        >
          {" "}
          <h5 className="card-title">
            {eventInfos.title}
          </h5>
          <h6 className="card-subtitle mb-3">
            {dayjs(eventInfos.date).format("HH:mm")} • {eventInfos.location} • {eventInfos.get_group_name}
          </h6>
        </a>
        <p className="card-subtitle mb-0">
          <ParticipateButton
            number_of_participants={eventInfos.number_of_participants}
            urls={urls}
            eventInfos={eventInfos}
          />
        </p>
        {/* <Truncate lines={3}>
          <p
            className="card-text"
            dangerouslySetInnerHTML={{ __html: eventInfos.description }}
          ></p>
        </Truncate> */}
      </div>
    </div>
  );
}
