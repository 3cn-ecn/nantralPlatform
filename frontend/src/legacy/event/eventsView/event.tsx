import * as React from 'react';
import { Card } from 'react-bootstrap';
import { EventInfos } from './interfaces';
import { getDate, getHour } from './utils';

interface EventProps {
  eventInfos: EventInfos;
}

export function Event({ eventInfos }: EventProps): JSX.Element {
  return (
    <Card border="#dc3545" className="mb-2">
      <Card.Img
        variant="top"
        src={eventInfos.image}
        onClick={() => {
          window.location.href = `/event/${eventInfos.id}/`;
        }}
        style={{ cursor: 'pointer' }}
      />
      <Card.Body as="a" href={`/event/${eventInfos.id}/`} className="pb-0 ">
        <small className="text-uppercase card-date">
          {getDate(eventInfos.start_date)}
        </small>
        <Card.Title as="h4">{eventInfos.title}</Card.Title>
        <Card.Subtitle className="mb-3">
          {[
            getHour(eventInfos.start_date),
            eventInfos.location,
            eventInfos.group.name,
          ].join(' • ')}
        </Card.Subtitle>
      </Card.Body>
    </Card>
  );
}
