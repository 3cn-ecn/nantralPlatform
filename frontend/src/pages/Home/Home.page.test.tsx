import { screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { EventDTO } from '#modules/event/infra/event.dto';
import { GroupPreviewDTO } from '#modules/group/infra/group.dto';
import { PostDTO } from '#modules/post/infra/post.dto';
import { mockServer } from '#shared/testing/mockServer';
import { renderWithProviders } from '#shared/testing/renderWithProviders';

import HomePage from './Home.page';

// mock ckeditor: replaced by #shared/ckeditor/__mocks__/CustomEditor.ts
jest.mock('#shared/ckeditor/CustomEditor.ts');

function mockPostApiCall(
  params: { pinned?: boolean; page?: number; page_size?: number },
  results: PostDTO[] = [],
) {
  mockServer
    .get('/api/post/post/')
    .query({ pinned: false, page: 1, page_size: 3, ...params })
    .reply(200, { count: results.length, results });
}

function mockEventApiCall(results: EventDTO[] = []) {
  mockServer
    .get('/api/event/event/')
    .query({
      from_date: /.*/,
      page: 1,
      page_size: /.*/,
    })
    .reply(200, { count: results.length, results });
}

function mockGroupApiCall(results: GroupPreviewDTO[]) {
  mockServer
    .get('/api/group/group/')
    .query({
      type: /.*/,
      is_member: true,
      is_admin: /.*/,
      slug: /.*/,
      search: /.*/,
      page: /.*/,
      page_size: /.*/,
      parent: /.*/,
    })
    .reply(200, { count: results.length, results });
}

describe('Home page', () => {
  it('should render correctly without data', async () => {
    // mock the api (since the backend does not exist during tests)
    mockEventApiCall([]);
    mockPostApiCall({ pinned: true }, []);
    mockPostApiCall({ pinned: false }, []);
    mockGroupApiCall([]);

    const component = renderWithProviders(<HomePage />);

    // At this point the page is still loading, so we use findBy and await
    // to wait for the element to appear.
    // We use findByRole to check that the element is a heading.
    // We use expect to make the test fails if the element is not found.
    expect(await screen.findByRole('heading', { name: 'Nantral Platform' }));

    // Now that the page is loaded, we check that the "Featured" section
    // (or "A la une" in French) is hidden, since we do not have any posts.
    expect(
      screen.queryByRole('heading', { name: 'Featured' }),
    ).not.toBeInTheDocument();

    // On the contrary, the Events section must be displayed
    expect(
      screen.queryByRole('heading', { name: 'Upcoming events' }),
    ).toBeInTheDocument();
    expect(screen.queryByText(/No upcoming event.../)).toBeInTheDocument();

    // finally, make a screenshot
    expect(component.asFragment()).toMatchSnapshot();
  });

  it('navigates to the event page when clicking on the "See all" button', async () => {
    mockEventApiCall([]);
    mockPostApiCall({ pinned: true }, []);
    mockPostApiCall({ pinned: false }, []);
    mockGroupApiCall([]);

    // create a fake event page for testing
    const fakeEventRoute = { path: '/event', element: <p>Events</p> };

    renderWithProviders(<HomePage />, [fakeEventRoute]);

    // click the 'See all' button
    const seeAllLink = screen.getAllByText('See all')[0]; // NB: we use get because we are not in an expect
    userEvent.click(seeAllLink);

    // check that we are on the event page
    expect(await screen.findByText('Events')).toBeInTheDocument();
  });
});
