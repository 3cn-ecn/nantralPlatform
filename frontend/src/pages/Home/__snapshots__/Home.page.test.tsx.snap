// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page should render correctly without data 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-l6ct9h"
  >
    <div
      class="mui-image-wrapper "
      style="width: 100%; height: 100%; display: flex; justify-content: center; align-items: center; background-color: inherit; position: absolute; z-index: -1;"
    >
      <img
        alt=""
        class="mui-image-img  css-jlmoib"
        src="/static/img/header.png"
        style="position: relative; width: 100%; height: 100%; object-fit: cover; transition-property: top, opacity; transition-duration: 900ms, 450ms; transition-timing-function: ease-in; opacity: 0; top: 100px;"
      />
    </div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1csjap5-MuiContainer-root"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 css-1pfj0oh-MuiTypography-root"
      >
        Welcome to
      </h4>
      <h1
        class="MuiTypography-root MuiTypography-poster css-1be4yq0-MuiTypography-root"
      >
        Nantral Platform
      </h1>
    </div>
  </div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1f2kw6w-MuiContainer-root"
  >
    <div
      class="MuiSpeedDial-root MuiSpeedDial-directionUp css-1mv8y2l-MuiSpeedDial-root"
      role="presentation"
    >
      <button
        aria-controls="SpeedDial-actions"
        aria-expanded="false"
        aria-haspopup="true"
        aria-label="SpeedDial"
        class="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeLarge MuiFab-primary MuiFab-root MuiFab-circular MuiFab-sizeLarge MuiFab-primary MuiSpeedDial-fab css-1qnbyf7-MuiButtonBase-root-MuiFab-root-MuiSpeedDial-fab"
        style="transform: none; webkit-transition: transform 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiSpeedDialIcon-root css-ohncnb-MuiSpeedDialIcon-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSpeedDialIcon-icon css-i4bv87-MuiSvgIcon-root"
            data-testid="AddIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <div
        aria-orientation="vertical"
        class="MuiSpeedDial-actions MuiSpeedDial-actionsClosed css-1s28u9o-MuiSpeedDial-actions"
        id="SpeedDial-actions"
        role="menu"
      >
        <span
          class="MuiSpeedDialAction-staticTooltip MuiSpeedDialAction-tooltipPlacementLeft MuiSpeedDialAction-staticTooltipClosed css-12w19fr-MuiSpeedDialAction-staticTooltip"
          id="SpeedDial-action-0"
        >
          <span
            class="MuiSpeedDialAction-staticTooltipLabel css-1b7vtjs-MuiSpeedDialAction-staticTooltipLabel"
            id="SpeedDial-action-0-label"
            style="transition-delay: 60ms;"
          >
            Event
          </span>
          <button
            aria-labelledby="SpeedDial-action-0-label"
            class="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default MuiSpeedDialAction-fab MuiSpeedDialAction-fabClosed css-rw3h6t-MuiButtonBase-root-MuiFab-root-MuiSpeedDialAction-fab"
            role="menuitem"
            style="transition-delay: 60ms;"
            tabindex="-1"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="EventIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M17 12h-5v5h5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1zm3 18H5V8h14z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </span>
        <span
          class="MuiSpeedDialAction-staticTooltip MuiSpeedDialAction-tooltipPlacementLeft MuiSpeedDialAction-staticTooltipClosed css-12w19fr-MuiSpeedDialAction-staticTooltip"
          id="SpeedDial-action-1"
        >
          <span
            class="MuiSpeedDialAction-staticTooltipLabel css-1b7vtjs-MuiSpeedDialAction-staticTooltipLabel"
            id="SpeedDial-action-1-label"
            style="transition-delay: 30ms;"
          >
            Post
          </span>
          <button
            aria-labelledby="SpeedDial-action-1-label"
            class="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default MuiSpeedDialAction-fab MuiSpeedDialAction-fabClosed css-rw3h6t-MuiButtonBase-root-MuiFab-root-MuiSpeedDialAction-fab"
            role="menuitem"
            style="transition-delay: 30ms;"
            tabindex="-1"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="PostAddIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M17 19.22H5V7h7V5H5c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2v-7h-2z"
              />
              <path
                d="M19 2h-2v3h-3c.01.01 0 2 0 2h3v2.99c.01.01 2 0 2 0V7h3V5h-3zM7 9h8v2H7zm0 3v2h8v-2h-3zm0 3h8v2H7z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </span>
      </div>
    </div>
    <div
      class="MuiBox-root css-i3pbo"
    >
      <div
        class="MuiBox-root css-47fqa9"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 css-2raco3-MuiTypography-root"
        >
          Upcoming events
        </h2>
        <a
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-disableElevation MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-disableElevation css-lvx8jz-MuiButtonBase-root-MuiButton-root"
          href="/event"
          tabindex="0"
        >
          See all
          <span
            class="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ChevronRightIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
      </div>
      <p
        class="MuiTypography-root MuiTypography-body1 css-17s6y1p-MuiTypography-root"
      >
        No upcoming event...¬†ü•∫ Create a new one!
      </p>
    </div>
    <div
      class="MuiBox-root css-i3pbo"
    >
      <div
        class="MuiBox-root css-47fqa9"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 css-2raco3-MuiTypography-root"
        >
          Posts
        </h2>
      </div>
      <p
        class="MuiTypography-root MuiTypography-body1 css-17s6y1p-MuiTypography-root"
      >
        No post in the last few days...¬†ü•∫ Create a new one!
      </p>
    </div>
    <div
      class="MuiBox-root css-c7yo1x"
    />
    <div
      class="MuiBox-root css-i3pbo"
    >
      <div
        class="MuiBox-root css-47fqa9"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 css-2raco3-MuiTypography-root"
        >
          Do you like this site?
        </h2>
      </div>
      <ul>
        <li>
          Level 1 ‚≠êÔ∏è: 
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-18z7zy0-MuiTypography-root-MuiLink-root"
            href="https://play.google.com/store/apps/details?id=org.ecn_3cn.nantral_platform"
          >
            Give us 5 stars on the Play Store!
          </a>
        </li>
        <li>
          Level 2 ü•∞: 
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-18z7zy0-MuiTypography-root-MuiLink-root"
            href="https://github.com/3cn-ecn/nantralPlatform"
          >
            Give us a star on Github!
          </a>
        </li>
        <li>
          Level 3 üòç: 
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-18z7zy0-MuiTypography-root-MuiLink-root"
            href="/suggestions"
          >
            Suggest improvements
          </a>
        </li>
        <li>
          Level 4 ü§©: 
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-18z7zy0-MuiTypography-root-MuiLink-root"
            href="https://docs.nantral-platform.fr/"
          >
            Help to code!
          </a>
        </li>
      </ul>
    </div>
    <div
      class="MuiBox-root css-y5dq5f"
    />
  </div>
</DocumentFragment>
`;
